common/deadlock/deadlock.go

```diff
@@ -154,9 +154,11 @@ func (lc *loopContext) run(ctx context.Context) error {
 		// unbuffered channel).
 		lc.ping(ctx, []common.Pingable{lc.root})
 
+		timer := time.NewTimer(lc.dd.config.Interval())
 		select {
-		case <-time.After(lc.dd.config.Interval()):
+		case <-timer.C:
 		case <-ctx.Done():
+			timer.Stop()
```

Comment:
Do we need to do this?

`// To ensure the channel is empty after a call to Stop, check the
// return value and drain the channel.
// For example, assuming the program has not received from t.C already:
//
//	if !t.Stop() {
//		<-t.C
//	}`

Reply:
I believe that timer channels are buffered with a capacity of one, so they don't need to be drained since there will be no blocked sender. However, I don't have a canonical source on this. I'm asking for one [here](https://github.com/golang/go/issues/27169#issuecomment-1629561455)